plugins/cassandra.yaml

```diff
@@ -0,0 +1,112 @@
+version: 0.0.1
+title: Apache Cassandra
+description: Log parser for Appache Cassandra
+parameters:
+  system_log_path:
+    label: Apache Cassandra System Log Path
+    description: Path to the Apache Cassandra system log file.
+    type: string
+    required: true
+  debug_log_path:
+    label: Apache Cassandra Debug Log Path
+    description: Path to the Apache Cassandra debug log file.
+    type: string
+  gc_log_path:
+    label: Apache Cassandra Garbage Collection Log Path
+    description: Path to the Apache Cassandra garbage collection log file.
+    type: string
+  start_at:
+    label: Start At
+    description: "Start reading file from 'beginning' or 'end'"
+    type: enum
+    valid_values:
+     - beginning
+     - end
+    default: end
+pipeline:
+  # {{ if .system_log_path }}
+  - id: system_log_reader
+    type: file_input
+    include:
+      - {{ .system_log_path }}
+    multiline:
+      line_start_pattern: '[A-Z]+\s+\[[^\]]+\]'
+    start_at: {{ or .start_at "end" }}
+
+  # Add log_name field to identify the log type  
+  - id: add_system_log_name
+    type: restructure
```

Comment:
Any reason why we aren't using the metadata plugin to accomplish this?

Reply:
No reason other than personal preference. Only reason to use metadata is for tags. Plus, metadata adds in EXPR($record._HOSTNAME) which is bit odd as that is the only plugin you have to use EXPR() I believe. Also, I can also do more with restructure should I need to say remove a field. It just reduces the number of operators needed in those cases but might make sense to use metadata operator when extra functionality of restructure isn't required.